### PR TITLE
{phys}[foss/2020a] Clang-Python-bindings v10.0.1, TRIQS v3.0.0, TRIQS-dft_tools v3.0.0, ... w/ Python 3.8.2

### PR DESCRIPTION
**Pull request by rokzitko**
(created using `eb --new-pr`)

### CODE REVIEW

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Review by boegel — DISMISSED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 6 (6 easyconfigs in this PR)
node2622.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/66c8909d500c1d841974938db22bd299 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 6 (6 easyconfigs in this PR)
node3148.skitty.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/4f859a754a4b4975165d4f5029a112b8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node3401.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/053117e3f7663cc2b5eec177573723e9 for a full test report.

**Comment by boegel** [comment]
@rokzitko Ignore the failing test reports, they were run in an environment (Slurm interactive job) in which MPI commands fail, it has nothing to do with the installation. I'll submit new test reports from the proper environment...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2622.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/14ce82d4d340e694d6c14fb67e259b59 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node3148.skitty.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/6baaadcacf1cf3a304135944412c4e18 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@rokzitko Looks like your last commit breaks the installation of `TRIQS-3.0.0-foss-2020a-Python-3.8.2.eb`?
```
/shared/easybuilder/CentOS8/haswell/software/binutils/2.34-GCCcore-9.3.0/bin/ld.gold: error: cannot find -lcpp2py
collect2: error: ld returned 1 exit status
make[2]: *** [c++/triqs/CMakeFiles/triqs_py.dir/build.make:95: c++/triqs/libtriqs_py.so.3.0] Error 1
```

I'm seeing the same issue on other systems too (CentOS 7, Haswell)

**Comment by rokzitko** [comment]
This is unexpected. I'll reintroduce the h5py dependency to fix this, even though it's not strictly speaking required.

**Comment by Wentzell** [comment]
The failure should not be related to the commit e8d164a in fact.
It seems that the build environment contained a previous install of cpp2py and cmake detected it (c.f. [log](https://gist.github.com/boegelbot/2631f3825965d7dc7bbdf3bc381e6fb9#file-triqs-3-0-0-foss-2020a-python-3-8-2_partial-log-L58)) at /users/boegelbot/CentOS8/haswell/software/TRIQS/3.0.0-foss-2020a-Python-3.8.2

There seems to currently be an issue when cpp2py is detected in the system and then linked.
I am investigating this.

@rokzitko I suggest that you

- Use `-DBuild_Deps=Always` during cmake configuration to avoid finding cpp2py in the system
- Remove the dependency of h5py after that

**Comment by boegel** [comment]
@Wentzell Ah, so the problem is that there's an existing installation that is not wiped in time, and then stuff is picked up from the populated installation directory while building? (yaay CMake)

Maybe it's worth adding `rm -rf %(installdir)s && ` to `preconfigopts` to avoid that?

**Comment by Wentzell** [comment]
> @Wentzell Ah, so the problem is that there's an existing installation that is not wiped in time, and then stuff is picked up from the populated installation directory while building? (yaay CMake)

Yes, exactly!

> Maybe it's worth adding `rm -rf %(installdir)s && ` to `preconfigopts` to avoid that?

That sounds also reasonable, yes.

**Comment by rokzitko** [comment]
> Maybe it's worth adding rm -rf %(installdir)s && to preconfigopts to avoid that?
> 
> That sounds also reasonable, yes.
> 
Yes, this should solve the issue. I've tried configopts = "-DBuild_Deps=Always", but I've encountered some other issues. I'm testing the preconfigopts solution locally right now.

**Comment by rokzitko** [comment]
Ok, everything seems in order now, including --force'd rebuilds which were failing before.

File: easybuild/easyconfigs/t/TRIQS/TRIQS-3.0.0-foss-2020a-Python-3.8.2.eb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    ('Mako', '1.1.2')
+]
+
+builddependencies = [
+    ('CMake', '3.16.4')
+]
+
+separate_build_dir = True
+
+preconfigopts = "rm -rf %(installdir)s && "
```
Comment (boegel) [inline easybuild/easyconfigs/t/TRIQS/TRIQS-3.0.0-foss-2020a-Python-3.8.2.eb:41]: @rokzitko This looks a bit weird here without motivation, so how about also adding a comment above this line to clarify why this is done?

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 6 (6 easyconfigs in this PR)
node3407.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/77a613ff47141da4688bc2a5826c2dd1 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node3407.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/a6248b395d5d0bf69cbbd501972727d5 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2493.golett.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/535de0ea128d5e8255ee68a42d54192b for a full test report.

**Comment by boegel** [comment]
Going in, thanks @rokzitko!

**Pull request merged**